### PR TITLE
[SPARK-49078][SQL][FOLLOWUP] Clear unused showColumnsNotSupportedForV2TablesError in QueryCompilationErrors and add more uts

### PR DESCRIPTION
**Pull request by xuzifu666**
### What changes were proposed in this pull request?

Clear unused error methods in QueryCompilationErrors.

### Why are the changes needed?

Due to had support show columns in v2 table(https://github.com/apache/spark/pull/47568), so the check method is not needed any more.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

not need

### CODE REVIEW

**Comment by xuzifu666** [comment]
@yaooqinn PTAL

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, @xuzifu666 . Please use a `[FOLLOWUP]` tag for the original root cause JIRA, @xuzifu666 .

**Comment by xuzifu666** [comment]
> Hi, @xuzifu666 . Please use a `[FOLLOWUP]` tag for the original root cause JIRA, @xuzifu666 .

@dongjoon-hyun OK，had change title name with FOLLOWUP formart. PTAL

**Review by yaooqinn — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by LuciferYang** [comment]
There are some other `unused declarations` in `QueryCompilationErrors`, such as `invalidStringParameter`, `cannotSaveVariantIntoExternalStorageError`, and `indeterminateCollationError`. Can we clean them up together in this pr? @xuzifu666

**Comment by xuzifu666** [comment]
> an we clean them up together in this pr?


> There are some other `unused declarations` in `QueryCompilationErrors`, such as `invalidStringParameter`, `cannotSaveVariantIntoExternalStorageError`, and `indeterminateCollationError`. Can we clean them up together in this pr? @xuzifu666

OK，had removed cannotSaveVariantIntoExternalStorageError and indeterminateCollationError in QueryCompilationErrors，invalidStringParameter had removed before which was not contain in master branch. @LuciferYang

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 1]
```diff
@@ -2378,6 +2378,8 @@ class DataSourceV2SQLSuiteV1Filter
     val t = "testcat.ns1.ns2.tbl"
     withTable(t) {
       spark.sql(s"CREATE TABLE $t (id bigint, data string) USING foo")
+      checkAnswer(sql(s"SHOW COLUMNS FROM $t IN testcat.ns1.ns2"), Seq(Row("id"), Row("data")))
```
Comment (LuciferYang) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2381]: Are the changes in these two lines related to the current PR?

Comment (xuzifu666) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2381]: Yes，from the last pr https://github.com/apache/spark/pull/47568， we lost the two line check，so this pr add these.

**Comment by LuciferYang** [comment]
https://github.com/apache/spark/blob/e893b32196afbf4b8e620239eb7aa61b38747241/sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala#L206-L211


> invalidStringParameter had removed before which was not contain in master branch.


seems that ` invalidStringParameter` is still in the master?

**Comment by xuzifu666** [comment]
> `invalidStringParameter`

Oh yes，I rebase master and found it，remove it now，Thanks @LuciferYang

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2382]: I apologize for not having a clear understanding of the background before and providing misleading guidance. 

If this is a followup to SPARK-49078, then indeed we should only clean up `showColumnsNotSupportedForV2TablesError` in this PR and add these two new test checks. 

However, for these two new test checks, we should modify the PR title and description to make them more relevant. 

For other unused methods in `QueryCompilationErrors`, they should be addressed in a separate PR.

Comment (xuzifu666) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2382]: OK，no problem，would make a separate pr to do clean other case，then this pr the same with last？ WDYT @LuciferYang

Comment (xuzifu666) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2382]: Had recover as before~ @LuciferYang

**Review by LuciferYang — APPROVED** [review]

**Comment by xuzifu666** [comment]
@LuciferYang ut had success，could you give a final review or merge it？

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @xuzifu666 and all. Merged to master.

**Pull request closed without merging**